spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/RuntimeTemplate.java

```diff
@@ -131,11 +131,27 @@ public Object postToActuator(String appId, String instanceId, String endpoint, M
 	public void postToUrl(String appId, String instanceId, byte[] data, HttpHeaders headers) {
 		Assert.notNull(appUrlPostUriTemplate, "post endpoint not found");
 		String uri = appUrlPostUriTemplate.expand(appId, instanceId).getHref();
+		waitForUrl(uri, 30_000L);
 		HttpEntity<byte[]> entity = new HttpEntity<>(data, headers);
 		ResponseEntity<String> response = this.restTemplate.exchange(uri, HttpMethod.POST, entity, String.class);
-
 		if (!response.getStatusCode().is2xxSuccessful()) {
 			throw new RuntimeException("POST:exception:" + response.getStatusCode() + ":" + response.getBody());
 		}
 	}
+
+	private void waitForUrl(String uri, long timeout) {
+		// Check
+		final long startTime = System.currentTimeMillis();
```

Comment:
We can compute the max time by `startTime + timeout`  here once as it never changes and it will probably make the `while` condition more readable.